addons/sale/models/sale.py

```diff
@@ -402,6 +403,28 @@ def _onchange_commitment_date(self):
                 }
             }
 
+    @api.onchange('pricelist_id', 'order_line')
```

Comment:
The order_line dependency isn't needed, new order lines are by default computed with current pricelist.

Reply:
@Feyensv  Done.